lib/chef/knife/azure_base.rb

```diff
@@ -128,8 +128,7 @@ def validate_params!
       end
 
       # validates keys
-      def validate!(keys)
-        errors = []
+      def validate!(keys, errors)
```

Comment:
@timh , thanks for contribution.
We'll test this PR to make sure that changing `validate!` method doesn't affect any other part of the code.